core/state/trie_prefetcher.go

```diff
@@ -77,20 +97,62 @@ func newTriePrefetcher(db Database, root common.Hash, namespace string) *triePre
 		storageWasteMeter: metrics.GetOrRegisterMeter(prefix+"/storage/waste", nil),
 	}
 	go p.abortLoop()
+	go p.mainLoop()
 	return p
 }
 
-func (p *triePrefetcher) abortLoop() {
+func (p *triePrefetcher) mainLoop() {
 	for {
 		select {
-		case fetcher := <-p.abortChan:
-			fetcher.abort()
-		case <-p.closeChan:
-			// drain fetcher channel
+		case pMsg := <-p.prefetchChan:
+			fetcher := p.fetchers[pMsg.root]
+			if fetcher == nil {
+				fetcher = newSubfetcher(p.db, pMsg.root, pMsg.accountHash)
+				p.fetchersMutex.Lock()
```

Comment:
Yes, we can eliminate the `prefetchChan` and move the logic to the caller side, but it would make the code logic more complicated. 
Since there could several routine do schedule at the same time, mean while, close will be executed in other routine... We have to add the sync protect for subfetcher's create and delete.
It is much clean to keep subfetcher's lifecycle within a routine,